hpccm/templates/ConfigureCMake.py

```diff
@@ -0,0 +1,139 @@
+
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=invalid-name, too-few-public-methods
+
+"""Documentation TBD"""
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+from six.moves import shlex_quote
+
+import logging  # pylint: disable=unused-import
+import os
+
+
+class ConfigureCMake(object):
```

Comment:
Is `ConfigureCMake` the right name for this template?  I assume this is inherited from `ConfigureMake`, but in that case the workflow is `configure && make`.  

There is an existing building block named `cmake`, so it could be confusing / lead to conflicts if this were also named `cmake`.  Ideas?

Reply:
Maybe `CMakeProject` or `CMakeBuild`?

Reply:
OK with either.  Slightly prefer `CMakeBuild`.